README.md

```diff
@@ -566,16 +566,18 @@ rather than applying them on the element yourself to avoid your props being
 overridden (or overriding the props returned). For example:
 `getInputProps({onKeyUp(event) {console.log(event)}})`.
 
+You can optionally provide a `mapStateToProps` function that provides you with access to some Downshift internals. This function will be called with `stateAndHelpers` and any props that are returned will be shallow merged with the standard props provided by the Prop Getter.
+
 <!-- This table was generated via http://www.tablesgenerator.com/markdown_tables -->
 
-| property               | type              | description                                                                                    |
-| ---------------------- | ----------------- | ---------------------------------------------------------------------------------------------- |
-| `getToggleButtonProps` | `function({})`    | returns the props you should apply to any menu toggle button element you render.               |
-| `getInputProps`        | `function({})`    | returns the props you should apply to the `input` element that you render.                     |
-| `getItemProps`         | `function({})`    | returns the props you should apply to any menu item elements you render.                       |
-| `getLabelProps`        | `function({})`    | returns the props you should apply to the `label` element that you render.                     |
-| `getMenuProps`         | `function({},{})` | returns the props you should apply to the `ul` element (or root of your menu) that you render. |
-| `getRootProps`         | `function({},{})` | returns the props you should apply to the root element that you render. It can be optional.    |
+| property               | type                               | description                                                                                                                                                                           |
+| ---------------------- | ---------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `getToggleButtonProps` | `function({}, mapStateToProps)`    | returns the props you should apply to any menu toggle button element you render.                                                                                                      |
```

Comment:
I'd really prefer the `getToggleButtonProps`, `getInputProps`, `getLabelProps`, `getMenuProps`, and `getRootProps` to accept an object _or_ `mapStateToProps` as the first argument. I'm fine with `getItemProps` requiring an object as the first argument and being slightly different than the rest. I'm thinking 99% of the time people will want one or the other but not both, so let's optimize for that case.